pysindy/optimizers/trapping_sr3.py

```diff
@@ -0,0 +1,516 @@
+                    # default solver is SCS here
+                    prob_m.solve(eps=self.eps_solver)  # , verbose=True)
+                    # end = time.time()
+                    # print('Total cvxpy time = ', end - start)
+                    # print("\nThe optimal m value is", prob_m.value)
+                    m = m_cp.value
+                    if m is None:
+                        print("Infeasible solve over m, increase/decrease eta")
+                        break
+            self.history_.append(coef_sparse.T)
+
+            if self.relax_optim:
+                # prox-grad for (A, m)
+                # Accelerated prox gradient descent
+                if self.accel:
+                    tk = (1 + np.sqrt(1 + 4 * tk_prev ** 2)) / 2.0
+                    m_partial = m + (tk_prev - 1.0) / tk * (m - m_prev)
+                    tk_prev = tk
+                    mPQ = np.tensordot(m_partial, PQ, axes=([0], [0]))
+                    p = PL - mPQ
+                    Pmatrix = p.reshape(r * r, r * N)
+                PW = np.tensordot(p, coef_sparse, axes=([3, 2], [0, 1]))
+                PQW = np.tensordot(PQ, coef_sparse, axes=([4, 3], [0, 1]))
+                A_b = (A - PW) / self.eta
+                PQWT_PW = np.tensordot(PQW, A_b, axes=([2, 1], [0, 1]))
+                # PQWT_PW = np.tensordot(PQW.T, A_b, axes=([2, 1], [0, 1]))
+                if self.accel:
+                    m = m_partial - self.alpha_m * PQWT_PW
+                else:
+                    m = m_prev - self.alpha_m * PQWT_PW
+                m_prev = m
+
+                # Update A
+                A = self._update_A(A - self.alpha_A * A_b, PW)
+            else:
+                PW = np.tensordot(p, coef_sparse, axes=([3, 2], [0, 1]))
+
+            # (m,A) update finished, append the result
+            self.m_history_.append(m)
+            self.A_history_.append(A)
+            eigvals, eigvecs = np.linalg.eig(PW)
+            self.PW_history_.append(PW)
+            self.PWeigs_history_.append(np.sort(eigvals))
+
+            # update objective
+            objective_history.append(self._objective(x, y, coef_sparse, A, PW, q))
+            q = q + 1
+
+            if (
+                self._m_convergence_criterion() < self.mtol
+                and self._convergence_criterion() < self.tol
+            ):
+                # Could not (further) select important features
+                break
+        else:
+            warnings.warn(
+                "trappingSR3._reduce did not converge after {} iterations.".format(
+                    self.max_iter
+                ),
+                ConvergenceWarning,
+            )
+
+        if self.use_constraints and self.constraint_order.lower() == "target":
+            self.constraint_lhs = reorder_constraints(
+                self.constraint_lhs, n_features, output_order="target"
+            )
+
+        self.coef_ = coef_sparse.T
+        self.objective_history = objective_history
```

Comment:
Please add this to the attributes section of the docstring too.

Reply:
Did this today